File: python/projects/non-public/threshold_tools/threshold_tools.py


class ThresholdTools:
    threshold_value: 0

    def __init__(self, tv):
        """
        Threshold Tools
        ---
        toolkit for comparing a process variable to a threshold
        value specified at the time of instantiation of the
        class.

        Parameters:
        tv (int) number the specified threshold value

        Returns:
        void
        """
        self.threshold_value = tv

    def at_or_above_threshold(self, pv):
        """
        At Or Above Threshold
        ---
        check that a process variable is at or above a given
        threshold value stored within this class instance.
        Parameters:
        pv (int) the process value under evaluation

        Returns:
        boolean
        """
        return pv >= self.threshold_value

    def update_threshold_value(self, tv):
        """
        Update Threshold Value
        ---
        update the locally maintained threshold value in this
        class instance to be compared against process variables.

        Parameters:
        tv (int) the new threshold value

        Returns:
        void
        """
        self.threshold_value = tv

File: typescript/projects/work/threshold-tools/threshold-tools.ts

/**
 * Threshold Tools
 * ---
 * A set of tooling to handle the ability to evaluate
 * a process variable against a threshold in a variety of
 * scenarios.
 */
export class ThresholdTools {
    thresholdValue: number;

    /**
     * At Or Above Threshold
     * ---
     * check that a process variable is at or above a given
     * threshold value stored within this class instance.
     * @param pv process variable under evaluation.
     */
    atOrAboveThreshold(pv: number): boolean {
        return pv >= this.thresholdValue;
    }


    constructor(tv: number) {
        this.thresholdValue = tv;
    }
}